bedrock/2_SignalsAndQueries/workflows.py

```diff
@@ -0,0 +1,110 @@
+import asyncio
+from datetime import timedelta
+from temporalio import workflow
+from collections import deque
+from typing import List, Tuple, Deque, Optional
+
+with workflow.unsafe.imports_passed_through():
+    from activities import prompt_bedrock
+
+
+@workflow.defn
+class SignalQueryBedrockWorkflow:
+    def __init__(self) -> None:
+        # List to store prompt history
+        self.conversation_history: List[Tuple[str, str]] = (
+            []
+        )
+        self.prompt_queue: Deque[str] = deque()
+        self.conversation_summary: str = ""
+
+    @workflow.run
+    async def run(self, inactivity_timeout_minutes: int) -> str:
+        while True:
+            workflow.logger.info(
+                "Waiting for prompts... or closing chat after " +
+                f"{inactivity_timeout_minutes} minute(s)"
+            )
+
+            # Wait for a chat message (signal) or timeout
+            try:
+                await workflow.wait_condition(
+                    lambda:
+                        bool(self.prompt_queue),
+                        timeout=timedelta(minutes=inactivity_timeout_minutes),
+                )
+            # if timeout was reached
+            except asyncio.TimeoutError:
+                workflow.logger.info("Chat closed due to inactivity")
+                # End the workflow
+                break
```

Comment:
If we've reached timeout, and are running final activities to close out the workflow, are signals that are received but not processed considered lost? What should be the expected behaviour?

Reply:
> are signals that are received but not processed considered lost?

Usually, since they might as well not be received if you never do anything with them. Maybe that is the behavior you want, but it's not the common behavior. Usually people want to process every signal they get. Closing something due to inactivity when there is activity right at the end is not usually good behavior, you should consider signals coming in anytime before actual close as the workflow no longer being "inactive".